_posts/2017-11-03-adapter-model.md

```diff
@@ -0,0 +1,83 @@
+<figcaption>Attribute Machine</figcaption></figure>
+
+Configuration is a complex task. In fact, evidence shows that the overwhelming majority of service outages are caused by configuration errors. To help combat this, Mixer’s configuration model enforces a number of constraints designed to avoid errors. For example, the configuration model uses strong typing to ensure that only meaningful attributes or attribute expressions are used in any given context.
+
+## Handlers: Configuring Adapters
+
+Each adapter that Mixer uses requires some configuration to operate. Typically, adapters need things like the URL to their backend, credentials, caching options, and so forth. Each adapter defines the exact configuration data it needs via a [protobuf](https://developers.google.com/protocol-buffers/) message.
+
+You configure each adapter by creating [*handlers*](https://istio.io/docs/concepts/policy-and-control/mixer-config.html#handlers) for them. A handler is a 
+configuration resource which represents a fully configured adapter ready for use. There can be any number of handlers for a single adapter, making it possible to reuse an adapter in different scenarios.
+
+## Templates: Adapter Input Schema
+
+Mixer is typically invoked twice for every incoming request to a mesh service, once for precondition checks and once for telemetry reporting. For every such call, Mixer invokes one or more adapters. Different adapters need different pieces of data as input in order to do their work. A logging adapter needs a log entry, a metric adapter needs a metric, an authorization adapter needs credentials, etc.
+Mixer [*templates*](https://istio.io/docs/reference/config/mixer/template/) are used to describe the exact data that an adapter consumes at request time.
+
+Each template is specified as a [protobuf](https://developers.google.com/protocol-buffers/) message. A single template describes a bundle of data that is delivered to one or more adapters at runtime. Any given adapter can be designed to support any number of templates, the specific templates the adapter supports is determined by the adapter developer. 
+
+[metric](https://istio.io/docs/reference/config/mixer/template/metric.html) and [logentry](https://istio.io/docs/reference/config/mixer/template/logentry.html) are two of the most essential templates used within Istio. They represent respectively the payload to report a single metric and a single log entry to appropriate backends. 
+
+## Instances: Attribute Mapping
+
+You control which data is delivered to individual adapters by creating [*instances*](https://istio.io/docs/concepts/policy-and-control/mixer-config
+.html#instances). Instances control how Mixer uses the [attributes](https://istio.io/docs/concepts/policy-and-control/attributes.html) delivered by the proxy into individual bundles of data that can be routed to different adapters.
```

Comment:
i think the line break here (between `mixer-config` and `.html`) screws up the markdown for the link.